tests/transformers/test_transformers_model_export.py

```diff
@@ -1926,6 +1934,84 @@ def test_table_question_answering_pyfunc_predict(table_question_answering_pipeli
     ]
 
 
+def test_feature_extraction_pipeline(feature_extraction_pipeline):
+    sentences = ["hi", "hello"]
+    signature = infer_signature(
+        sentences,
+        mlflow.transformers.generate_signature_output(feature_extraction_pipeline, sentences),
+    )
+
+    artifact_path = "feature_extraction_pipeline"
+    with mlflow.start_run():
+        model_info = mlflow.transformers.log_model(
+            transformers_model=feature_extraction_pipeline,
+            artifact_path=artifact_path,
+            signature=signature,
+            input_example=["A sentence", "Another sentence"],
+        )
+
+    # Load as native
+    loaded_pipeline = mlflow.transformers.load_model(model_info.model_uri)
+
+    inference_single = "Testing"
+    inference_mult = ["Testing something", "Testing something else"]
+
+    pred = loaded_pipeline(inference_single)
+    assert len(pred[0][0]) > 10
+    assert isinstance(pred[0][0][0], float)
+
+    pred_multiple = loaded_pipeline(inference_mult)
+    assert len(pred_multiple[0][0]) > 2
+    assert isinstance(pred_multiple[0][0][0][0], float)
+
+    loaded_pyfunc = mlflow.pyfunc.load_model(model_info.model_uri)
+
+    pyfunc_pred = loaded_pyfunc.predict(inference_single)
+
+    assert isinstance(pyfunc_pred, np.ndarray)
+
+    assert np.array_equal(np.array(pred[0]), pyfunc_pred)
+
+    pyfunc_pred_multiple = loaded_pyfunc.predict(inference_mult)
+
+    assert np.array_equal(np.array(pred_multiple[0][0]), pyfunc_pred_multiple)
+
+
+def test_feature_extraction_pipeline_pyfunc_predict(feature_extraction_pipeline):
+    artifact_path = "feature_extraction"
+    with mlflow.start_run():
+        model_info = mlflow.transformers.log_model(
+            transformers_model=feature_extraction_pipeline,
+            artifact_path=artifact_path,
+        )
+
+    inference_payload = json.dumps({"inputs": ["sentence one", "sentence two"]})
+
+    response = pyfunc_serve_and_score_model(
+        model_info.model_uri,
+        data=inference_payload,
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON,
+        extra_args=["--env-manager", "local"],
+    )
+    values = PredictionsResponse.from_json(response.content.decode("utf-8")).get_predictions()
+
+    assert len(values.columns) == 384
+    assert len(values) == 4
```

Comment:
What is the expected value 4?

Reply:
The encoder will generate a unique encoding vector for each tokenized element in the trained tokenizer based on the tokenizer and model configuration, regardless of the sentence count submitted. For example:

``` python
import transformers

st_arch = 'sentence-transformers/all-MiniLM-L6-v2'
model = transformers.AutoModel.from_pretrained(st_arch)
tokenizer = transformers.AutoTokenizer.from_pretrained(st_arch)

pipe = transformers.pipeline(model=model, tokenizer=tokenizer, task="feature-extraction")

sentences = [["hello", "happy"], 
             ["Generates a base flavor metadata collection for use in parsing the submitted values"],
             ["I have a rocket ship!", "rhombitruncatedicosododecahedron is the term for a 100-sided polyhedron"]
            ]
for sentence in sentences:
    print(sentence)
    output = pipe(*sentences)
    print(f"The length of the encoding vectors is: {len(output[0][0])}")
```

output:
``` bash
['hello', 'happy']
The length of the encoding vectors is: 3
['Generates a base flavor metadata collection for use in parsing the submitted values']
The length of the encoding vectors is: 3
['I have a rocket ship!', 'rhombitruncatedicosododecahedron is the term for a 100-sided polyhedron']
The length of the encoding vectors is: 3

```

Reply:
As a bit of side research, I'm planning on looking at the implementation details of those models (since I'm curious now). LMK if you find anything interesting as well ;)